In [1]:
import pandas as pd
import numpy as np

In [2]:
from selenium import webdriver

browser = webdriver.Chrome("D:/Arpan_Python_Programming/DataScraping & Data Wrangling/chromedriver.exe")

In [3]:
url = "https://www.moneycontrol.com/financials/interglobeaviation/consolidated-balance-sheetVI/IA04"
browser.get(url)

In [4]:
ind_balsheet = pd.read_html(url)

In [5]:
new_balsheet = ind_balsheet[2].iloc[6:81,]
    
new_balsheet.columns = ["Equity_shareholder", "MAR-19","MAR-18","MAR-17"]
new_balsheet.head()

Equity_shareholder   MAR-19   MAR-18   MAR-17
6       Equity Share Capital   384.41   384.41   361.47
7   Preference Share Capital     0.00     0.00     0.00
8        Total Share Capital   384.41   384.41   361.47
9       Revaluation Reserves     0.00     0.00     0.00
10      Reserves and Surplus  6561.40  6692.95  3417.71

In [12]:
url1 = "https://www.moneycontrol.com/financials/interglobeaviation/profit-lossVI/IA04#IA04"
browser.get(url1)

In [13]:
profit_loss = pd.read_html(url1)

In [19]:
new_profit = profit_loss[3].iloc[5:49,:4]
new_profit.columns = ["Income", "MAR-19","MAR-18","MAR-17"]
new_profit.head()

Income    MAR-19    MAR-18    MAR-17
5  Revenue From Operations [Gross]  28496.77  22500.52  18440.07
6    Revenue From Operations [Net]  28496.77  22500.52  18440.07
7         Other Operating Revenues      0.00    520.37    140.43
8         Total Operating Revenues  28496.77  23020.89  18580.50
9                     Other Income   1324.94    946.86    789.07

In [21]:
new_profit = new_profit.fillna(0)

In [22]:
new_balsheet = new_balsheet.fillna(0)

In [24]:
new_profit.to_csv("profit_loss.csv",index=False)

In [25]:
new_balsheet.to_csv("indigo_balancesheet.csv",index=False)

# |||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||||

In [2]:
bal = pd.read_csv("indigo_balancesheet.csv")
profit = pd.read_csv("profit_loss.csv")

In [3]:
bal

Equity_shareholder    MAR-19     MAR-18  \
0                                Equity Share Capital    384.41     384.41   
1                            Preference Share Capital      0.00       0.00   
2                                 Total Share Capital    384.41     384.41   
3                                Revaluation Reserves      0.00       0.00   
4                                Reserves and Surplus   6561.40    6692.95   
5                          Total Reserves and Surplus   6561.40    6692.95   
6               Money Received Against Share Warrants      0.00       0.00   
7                             Employees Stock Options      0.00       0.00   
8                            Total Shareholders Funds   6945.81    7077.36   
9    Preference Shares Issued By Subsidiary Companies      0.00       0.00   
10                     Equity Share Application Money      0.00       0.00   
11                 Preference Share Application Money      0.00       0.00   
12                             Share Capital Suspense      0.00       0.00   
13                       Hybrid/Debt/Other Securities      0.00       0.00   
14                        Statutory Consumer Reserves      0.00       0.00   
15         Special Appropriation Towards Project Cost      0.00       0.00   
16           Service Line Contribution From Consumers      0.00       0.00   
17                            Government/Other Grants      0.00       0.00   
18                                  Minority Interest      0.00       0.00   
19                               Policy Holders Funds      0.00       0.00   
20                      Group Share In Joint Ventures      0.00       0.00   
21                            NON-CURRENT LIABILITIES      0.00       0.00   
22                               Long Term Borrowings   2193.67    2241.37   
23                     Deferred Tax Liabilities [Net]     64.42     369.53   
24                        Other Long Term Liabilities   7536.77    5121.12   
25                               Long Term Provisions    275.82     196.89   
26                      Total Non-Current Liabilities  10070.68    7928.91   
27  Foreign Currency Monetary Item Translation Dif...      0.00       0.00   
28                                CURRENT LIABILITIES      0.00       0.00   
29                              Short Term Borrowings      0.00       0.00   
..                                                ...       ...        ...   
45                                       Fixed Assets   5685.67    4611.30   
46                            Non-Current Investments      0.02       0.02   
47                          Deferred Tax Assets [Net]      0.97       0.00   
48                       Long Term Loans And Advances    584.40     683.13   
49                           Other Non-Current Assets    638.75    1203.28   
50                           Total Non-Current Assets   6909.80    6497.73   
51                                  Minority Interest      0.00       0.00   
52                      Group Share In Joint Ventures      0.00       0.00   
53  Foreign Currency Monetary Item Translation Dif...      0.00       0.00   
54                                     CURRENT ASSETS      0.00       0.00   
55                                Current Investments   6516.57    6343.91   
56                                        Inventories    211.44     183.23   
57                                  Trade Receivables    362.47     226.32   
58                          Cash And Cash Equivalents   8622.00    6580.69   
59                      Short Term Loans And Advances    466.96     191.50   
60                                 OtherCurrentAssets   1937.61    1105.92   
61                               Total Current Assets  18117.05   14631.56   
62                                       Total Assets  25026.84   21129.30   
63                       OTHER ADDITIONAL INFORMATION      0.00       0.00   
64                CONTINGENT LIABILITIES, COMMITMENTS    

In [15]:
import numpy as np

bal = bal.replace({0:np.NaN})

In [16]:
bal = bal.dropna()
bal.head()

Equity_shareholder   MAR-19   MAR-18   MAR-17
0        Equity Share Capital   384.41   384.41   361.47
2         Total Share Capital   384.41   384.41   361.47
4        Reserves and Surplus  6561.40  6692.95  3417.71
5  Total Reserves and Surplus  6561.40  6692.95  3417.71
8    Total Shareholders Funds  6945.81  7077.36  3779.18

In [17]:
profit.head()

Income    MAR-19    MAR-18    MAR-17
0  Revenue From Operations [Gross]  28496.77  22500.52  18440.07
1    Revenue From Operations [Net]  28496.77  22500.52  18440.07
2         Other Operating Revenues      0.00    520.37    140.43
3         Total Operating Revenues  28496.77  23020.89  18580.50
4                     Other Income   1324.94    946.86    789.07

In [18]:
#dropping those row which is having 0 values
profit.drop([6,16,17,18,19,22,30,31,32,33,34,35,38,39,40],inplace=True)

In [19]:
profit

Income    MAR-19    MAR-18  \
0                     Revenue From Operations [Gross]  28496.77  22500.52   
1                       Revenue From Operations [Net]  28496.77  22500.52   
2                            Other Operating Revenues      0.00    520.37   
3                            Total Operating Revenues  28496.77  23020.89   
4                                        Other Income   1324.94    946.86   
5                                       Total Revenue  29821.71  23967.74   
7                          Purchase Of Stock-In Trade    139.80    123.88   
8                              Aircraft Fuel Expenses      0.00   7760.14   
9                              Aircraft Lease Rentals      0.00   3610.20   
10  Changes In Inventories Of FG,WIP And Stock-In ...     -0.66      1.27   
11                          Employee Benefit Expenses   3137.79   2455.02   
12                                      Finance Costs    508.96    339.82   
13             Depreciation And Amortisation Expenses    759.58    436.88   
14                                     Other Expenses  25425.28   6113.88   
15                                     Total Expenses  29970.76  20841.07   
20  Profit/Loss Before Exceptional, ExtraOrdinary ...   -149.05   3126.68   
21                             Profit/Loss Before Tax   -149.05   3126.68   
23                                        Current Tax   -305.18    668.98   
24                       Less: MAT Credit Entitlement      0.00      0.00   
25                                       Deferred Tax      0.00    215.32   
26                                 Total Tax Expenses   -305.18    884.30   
27  Profit/Loss After Tax And Before ExtraOrdinary...    156.14   2242.37   
28             Profit/Loss From Continuing Operations    156.14   2242.37   
29                         Profit/Loss For The Period    156.14   2242.37   
36                                    Basic EPS (Rs.)      4.06     60.03   
37                                  Diluted EPS (Rs.)      4.06     59.90   
41                              Equity Share Dividend      0.00   1229.73   
42                                    Tax On Dividend      0.00    250.34   
43                           Equity Dividend Rate (%)     50.00     60.00   

      MAR-17  
0   18440.07  
1   18440.07  
2     140.43  
3   18580.50  
4     789.07  
5   19369.57  
7     123.83  
8    6341.51  
9    3125.37  
10     -0.29  
11   2048.19  
12    330.78  
13    457.25  
14   4798.58  
15  17225.23  
20   2144.34  
21   2144.34  
23    491.15  
24      0.00  
25     -6.00  
26    485.15  
27   1659.19  
28   1659.19  
29   1659.19  
36     45.94  
37     45.85  
41    542.20  
42    110.38  
43    340.00

In [20]:
profit = profit.iloc[:,:]
profit["profit_loss"] = profit["Income"].map({"Total Revenue":"Turnover","Basic EPS (Rs.)":"EPS",
                                             "Profit/Loss For The Period":"Net-Profit"})
profit[["MAR-19","MAR-18","MAR-17"]] = profit[["MAR-19","MAR-18","MAR-17"]].astype(float)
profit = profit.groupby("profit_loss")["MAR-19","MAR-18","MAR-17"].sum().reset_index()
profit

profit_loss    MAR-19    MAR-18    MAR-17
0         EPS      4.06     60.03     45.94
1  Net-Profit    156.14   2242.37   1659.19
2    Turnover  29821.71  23967.74  19369.57

In [21]:
EPS = profit.iloc[0,1]
NP = profit.iloc[1,1]
Turnover = profit.iloc[2,1]

EPS_18 = profit.iloc[0,2]
NP_18 = profit.iloc[1,2]
Turnover_18 = profit.iloc[2,2]

EPS_17 = profit.iloc[0,3]
NP_17 = profit.iloc[1,3]
Turnover_17 = profit.iloc[2,3]

In [22]:
bal["asset_liabilities"] = bal["Equity_shareholder"].map({"Total Share Capital":"NW","Total Reserves and Surplus":"NW",
                                                         "Long Term Borrowings":"LTL","Other Long Term Liabilities":"LTL","Long Term Provisions":"LTL","Long Term Loans And Advances":"LTL",
                                                         "Total Current Liabilities":"CL","Total Non-Current Liabilities":"NCL",
                                                         "Tangible Assets":"TA","Fixed Assets":"FA","Total Non-Current Assets":"NCA",
                                                         "Total Current Assets":"CA","Total Assets":"TA","Cash And Cash Equivalents":"QA","Inventories":"INVENT",
                                                         "Equity Share Capital":"Share_Cap","Reserves and Surplus":"Resrv_Surpls"})
                                                          

In [23]:
bal

Equity_shareholder    MAR-19    MAR-18    MAR-17  \
0             Equity Share Capital    384.41    384.41    361.47   
2              Total Share Capital    384.41    384.41    361.47   
4             Reserves and Surplus   6561.40   6692.95   3417.71   
5       Total Reserves and Surplus   6561.40   6692.95   3417.71   
8         Total Shareholders Funds   6945.81   7077.36   3779.18   
22            Long Term Borrowings   2193.67   2241.37   2395.71   
23  Deferred Tax Liabilities [Net]     64.42    369.53    161.81   
24     Other Long Term Liabilities   7536.77   5121.12   3966.02   
25            Long Term Provisions    275.82    196.89    122.39   
26   Total Non-Current Liabilities  10070.68   7928.91   6645.93   
30                  Trade Payables   1452.81   1000.20    774.61   
31       Other Current Liabilities   6393.72   5019.59   3943.31   
32           Short Term Provisions    163.83    103.25     66.71   
33       Total Current Liabilities   8010.36   6123.03   4784.63   
34   Total Capital And Liabilities  25026.84  21129.30  15209.74   
37                 Tangible Assets   5685.67   4534.76   3747.47   
45                    Fixed Assets   5685.67   4611.30   3819.03   
46         Non-Current Investments      0.02      0.02      0.02   
48    Long Term Loans And Advances    584.40    683.13    544.03   
49        Other Non-Current Assets    638.75   1203.28   1400.27   
50        Total Non-Current Assets   6909.80   6497.73   5763.34   
55             Current Investments   6516.57   6343.91   3713.41   
56                     Inventories    211.44    183.23    163.15   
57               Trade Receivables    362.47    226.32    158.70   
58       Cash And Cash Equivalents   8622.00   6580.69   4632.55   
59   Short Term Loans And Advances    466.96    191.50      3.98   
60              OtherCurrentAssets   1937.61   1105.92    774.61   
61            Total Current Assets  18117.05  14631.56   9446.39   
62                    Total Assets  25026.84  21129.30  15209.74   

   asset_liabilities  
0          Share_Cap  
2                 NW  
4       Resrv_Surpls  
5                 NW  
8                NaN  
22               LTL  
23               NaN  
24               LTL  
25               LTL  
26               NCL  
30               NaN  
31               NaN  
32               NaN  
33                CL  
34               NaN  
37                TA  
45                FA  
46               NaN  
48               LTL  
49               NaN  
50               NCA  
55               NaN  
56            INVENT  
57               NaN  
58                QA  
59               NaN  
60               NaN  
61                CA  
62                TA

In [24]:
long_term_bor = bal.iloc[22,1]
long_term_bor_18 = bal.iloc[22,2]
long_term_bor_17 = bal.iloc[22,3]

In [25]:
final_bal = bal.groupby("asset_liabilities")["MAR-19","MAR-18","MAR-17"].sum().reset_index()
final_bal

asset_liabilities    MAR-19    MAR-18    MAR-17
0                 CA  18117.05  14631.56   9446.39
1                 CL   8010.36   6123.03   4784.63
2                 FA   5685.67   4611.30   3819.03
3             INVENT    211.44    183.23    163.15
4                LTL  10590.66   8242.51   7028.15
5                NCA   6909.80   6497.73   5763.34
6                NCL  10070.68   7928.91   6645.93
7                 NW   6945.81   7077.36   3779.18
8                 QA   8622.00   6580.69   4632.55
9       Resrv_Surpls   6561.40   6692.95   3417.71
10         Share_Cap    384.41    384.41    361.47
11                TA  30712.51  25664.06  18957.21

In [26]:
CA = final_bal.iloc[0,1]
CL = final_bal.iloc[1,1]
FA = final_bal.iloc[2,1]
INVENT = final_bal.iloc[3,1]
LTL = final_bal.iloc[4,1]
NCA = final_bal.iloc[5,1]
NCL = final_bal.iloc[6,1]
NW = final_bal.iloc[7,1]
QA = final_bal.iloc[8,1]
Reserv_Surpls = final_bal.iloc[9,1]
Share_Cap = final_bal.iloc[10,1]
TA = final_bal.iloc[11,1]

In [27]:
CA_18 = final_bal.iloc[0,2]
CL_18 = final_bal.iloc[1,2]
FA_18 = final_bal.iloc[2,2]
INVENT_18 = final_bal.iloc[3,2]
LTL_18 = final_bal.iloc[4,2]
NCA_18 = final_bal.iloc[5,2]
NCL_18 = final_bal.iloc[6,2]
NW_18 = final_bal.iloc[7,2]
QA_18 = final_bal.iloc[8,2]
Reserv_Surpls_18 = final_bal.iloc[9,2]
Share_Cap_18 = final_bal.iloc[10,2]
TA_18 = final_bal.iloc[11,2]

In [28]:
CA_17 = final_bal.iloc[0,3]
CL_17 = final_bal.iloc[1,3]
FA_17 = final_bal.iloc[2,3]
INVENT_17 = final_bal.iloc[3,3]
LTL_17 = final_bal.iloc[4,3]
NCA_17 = final_bal.iloc[5,3]
NCL_17 = final_bal.iloc[6,3]
NW_17 = final_bal.iloc[7,3]
QA_17 = final_bal.iloc[8,3]
Reserv_Surpls_17 = final_bal.iloc[9,3]
Share_Cap_17 = final_bal.iloc[10,3]
TA_17 = final_bal.iloc[11,3]

## Ratio Analysis
1. current ratio
2. quick ratio
3. debt equity ratio
4. P E ratio
5. net profit margin ratio
6. return on equity
7. return on investment
8. return on capital employee
9. net profit/ sales

## RATIO ANALYSIS FOR-2019

In [30]:
current_ratio = CA/CL
print("1.Current Ratio: ",current_ratio)
print("\n")

quick_ratio = (QA-INVENT)/CL
print("2.Quick Ratio: ",quick_ratio)
print("\n")

IA = 0
TNW = NW - IA
debt_equity_ratio = LTL/TNW
print("3.Debt Equity Ratio: ",debt_equity_ratio)
print("\n")

#P/E Ratio = Earnings per share / Market value per share
#capital_employed = debt/(cap_emp=debt+equity)
capital_employed = long_term_bor+(Share_Cap+Reserv_Surpls)

debt_to_capital_employed = long_term_bor/capital_employed
print("4.Debt to Capital Employed Ratio: ",debt_to_capital_employed)
print("\n")

#ROCE = profit/capital_employed *100
ROCE = (NP*100)/capital_employed
print("5.Return on Capital Employed: ",ROCE)
print("\n")

#netprofit_margin = net_profit/turnover(revenu)
netprofit_margin = NP/Turnover
print("6.Net Profit Margin: ",netprofit_margin)
print("\n")

#ROE = net profit/ TNW
ROE = NP / TNW
print("7.Return on Equity: ",ROE)
print("\n")

#ROI = net profit / TNW + LTL
ROI = NP / (TNW + LTL)
print("8.Return on Interest: ",ROI)
print("\n")

Avg_total_asset = (CA+FA)/2
Sales_Turnover_Ratio = Turnover/Avg_total_asset
print("9.Sales Turnover Ratio: ",Sales_Turnover_Ratio)

1.Current Ratio:  2.2617023454626257


2.Quick Ratio:  1.0499603014096746


3.Debt Equity Ratio:  1.5247552121350858


4.Debt to Capital Employed Ratio:  0.029542072723462227


5.Return on Capital Employed:  2.1815641482412937


6.Net Profit Margin:  0.0052357829245874895


7.Return on Equity:  0.0224797395840082


8.Return on Interest:  0.008903730340256618


9.Sales Turnover Ratio:  2.5057396801710055


## RATIO ANALYSIS FOR-2018

In [32]:
current_ratio = CA_18/CL_18
print("1.Current Ratio: ",current_ratio)
print("\n")

quick_ratio = QA_18-INVENT_18/CL_18
print("2.Quick Ratio: ",quick_ratio)
print("\n")

IA = 0
TNW = NW_18 - IA
debt_equity_ratio = LTL_18/TNW
print("3.Debt Equity Ratio: ",debt_equity_ratio)
print("\n")

#P/E Ratio = Earnings per share / Market value per share
capital_employed = long_term_bor_18+(Share_Cap_18+Reserv_Surpls_18)

debt_to_capital_employed = long_term_bor_18/capital_employed
print("4.Debt to Capital Employed Ratio: ",debt_to_capital_employed)
print("\n")

#ROCE = profit/capital_employed *100
ROCE = (NP_18*100)/capital_employed
print("5.Return on Capital Employed: ",ROCE)
print("\n")

#netprofit_margin = net_profit/turnover(revenu)
netprofit_margin = NP_18/Turnover_18
print("6.Net Profit Margin: ",netprofit_margin)
print("\n")

#ROE = net profit/ TNW
ROE = NP_18 / TNW
print("7.Return on Equity: ",ROE)
print("\n")

#ROI = net profit / TNW + LTL
ROI = NP_18 / (TNW + LTL_18)
print("8.Return on Interest: ",ROI)
print("\n")

Avg_total_asset = (CA_18+FA_18)/2
Sales_Turnover_Ratio = Turnover_18/Avg_total_asset
print("9.Sales Turnover Ratio: ",Sales_Turnover_Ratio)

1.Current Ratio:  2.3895946941301935


2.Quick Ratio:  6580.660075273189


3.Debt Equity Ratio:  1.1646305967196808


4.Debt to Capital Employed Ratio:  0.025236241131919034


5.Return on Capital Employed:  30.884129251204108


6.Net Profit Margin:  0.0935578406641594


7.Return on Equity:  0.31683706918964133


8.Return on Interest:  0.1463700409990424


9.Sales Turnover Ratio:  2.491078768956382


## RATIO ANALYSIS FOR-2017

In [33]:
current_ratio = CA_17/CL_17
print("1.Current Ratio: ",current_ratio)
print("\n")

quick_ratio = (QA_17-INVENT_17)/CL_17
print("2.Quick Ratio: ",quick_ratio)
print("\n")

IA = 0
TNW = NW_17 - IA
debt_equity_ratio = LTL_17/TNW
print("3.Debt Equity Ratio: ",debt_equity_ratio)
print("\n")

#P/E Ratio = Earnings per share / Market value per share
#capital_employed = debt/(cap_emp=debt+equity)
capital_employed = long_term_bor_17+(Share_Cap_17+Reserv_Surpls_17)

debt_to_capital_employed = long_term_bor_17/capital_employed
print("4.Debt to Capital Employed Ratio: ",debt_to_capital_employed)
print("\n")

#ROCE = profit/capital_employed *100
ROCE = (NP_17*100)/capital_employed
print("5.Return on Capital Employed: ",ROCE)
print("\n")

#netprofit_margin = net_profit/turnover(revenu)
netprofit_margin = NP_17/Turnover_17
print("6.Net Profit Margin: ",netprofit_margin)
print("\n")

#ROE = net profit/ TNW
ROE = NP_17 / TNW
print("7.Return on Equity: ",ROE)
print("\n")

#ROI = net profit / TNW + LTL
ROI = NP_17 / (TNW + LTL_17)
print("8.Return on Interest: ",ROI)
print("\n")

Avg_total_asset = (CA_17+FA_17)/2
Sales_Turnover_Ratio = Turnover_17/Avg_total_asset
print("9.Sales Turnover Ratio: ",Sales_Turnover_Ratio)

1.Current Ratio:  1.9743198533637918


2.Quick Ratio:  0.9341161176517307


3.Debt Equity Ratio:  1.8597023692970431


4.Debt to Capital Employed Ratio:  0.04138415607014126


5.Return on Capital Employed:  42.08653258352294


6.Net Profit Margin:  0.08565961970245081


7.Return on Equity:  0.4390343937044544


8.Return on Interest:  0.15352450605283638


9.Sales Turnover Ratio:  2.9203100994917612


#                                         INSIGHT ABOUT INDIGO                  

**CURRENT RATIO:**
* current ratio ratio under 1 indicates that the company’s debts due in a year or less are greater than its assets 
the higher the current ratio, the more capable a company is of paying its obligations because it has a larger proportion
of short-term asset value relative to the value of its short-term liabilities. However, while a high ratio, say over 3, 
could indicate the company can cover its current liabilities three times, it may indicate that its not using its current
assets efficiently, is not securing financing very well, or is not managing its working capital.

* Indigo airliance current ratio for 2019,2018 and 2017 are 2.26,2.38 and 1.97 which indicates they are very well mannaging
their asset to the right place ,that means company can cover its current liabilities 2 times.

**QUICK RATIO:**
* 1 is considered to be the normal quick ratio, as it indicates that the company is fully equipped with exactly
  enough assets to be instantly liquidated to pay off its current liabilities. A company that has a quick ratio of less   than 1 may not be able to fully pay off its current liabilities in the short term, while a company having a quick ratio higher than 1 can instantly get rid of its current liabilities. For instance, a quick ratio of 1.5 indicates that the company has$1.50 of liquid assets available to cover each $1 of its current liabilities.

* Above company has quick ratio 1.04, 6580,0.93 in 2019,2018,2017 which means 17 they had less than 1 indicates the companyhas $0.93 of liquid assets available to cover each $1 of its current liabilities. through out the year it has improved.

**DEBT EQUITY RATIO:**
* In a basic sense, Total Debt / Equity is a measure of all of a company's future obligations on the balance sheet relative
  to equity. A high ratio may indicate that the company is much resourced with (outside) borrowing as compared to funding 
  from shareholders.
* debt equity ratio for INDIGO from 2019-2017 are 1.5, 1.16 and 1.85, which means quit good.

**DEBT TO CAPITAL EMPLOYEED RATIO:**
* The debt-to-capital ratio is a measurement of a company's financial leverage.The debt-to-capital ratio gives analysts and
  investors a better idea of a company's financial structure and whether or not the company is a suitable investment.
  All else being equal, the higher the debt-to-capital ratio, the riskier the company. 

**RETURN ON CAPITAL EMPLOYED**
* Return on capital employed (ROCE) is a financial ratio that measures a company's profitability and the efficiency with
  which its capital is used. In other words, the ratio measures how well a company is generating profits from its capital.
 
* ROCE= Capital Employed / EBIT
* EBIT=Earnings before interest and tax
* Capital Employed=Total assets − Current liabilities
* INDIGO from 2019,2018 and 2017 ROCE are 2.18,30,42 that means profit of the company in 2017 was better than the current
  year

**RETURN ON EQUITY:**
* ROE is considered a measure of how effectively management is using a company’s assets to create profits. ROE is expressed
  as a percentage and can be calculated for any company if net income and equity are both positive numbers.
* A normal ROE in the utility sector could be 10% or less. A technology or retail firm with smaller balance sheet accounts
  relative to net income may have normal ROE levels of 18% or more.Any ROE less than 10% is poor.
* 2%, 31% and 43% since from 2019 to 2017 performance of the company is going down with the year past.

**Return on Investment:**
* ROI tries to directly measure the amount of return on a particular investment, relative to the investment’s cost. 
* ROI = (Current Value of Investment - Cost of Investment) / Cost of Investment

**Net Profit Margin:**
* The net profit margin is equal to how much net income or profit is generated as a percentage of revenue. Net profit   margin
is the ratio of net profits to revenues for a company or business segment. Net profit margin is typically expressed as a
percentage but can also be represented in decimal form. The net profit margin illustrates how much of each dollar in 
revenue collected by a company translates into profit.

# \\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\\